service/storage_service.cc

```diff
@@ -2101,7 +2101,8 @@ future<> storage_service::join_token_ring(cdc::generation_service& cdc_gen_servi
     });
     _listeners.emplace_back(make_lw_shared(std::move(schema_change_announce)));
     co_await _gossiper.wait_for_gossip_to_settle();
-    co_await feature_enabler.enable_features_on_join(_gossiper);
+    // TODO: Look at the group 0 upgrade state and use it to decide whether to attach or not
+    co_await feature_enabler.enable_features_on_join(_gossiper, _raft_topology_change_enabled);
```

Comment:
I think it's simpler to do

```
    if (!_raft_topology_change_enabled) {
        co_await feature_enabler.enable_features_on_join(_gossiper);
    }
```

and avoid argument to the `.enable_features_on_join()`

Reply:
Agree, I'll change it.